### PR TITLE
AsyncHttpClient: return from requests before body completion

### LINKED ISSUE #8109: async httpclient blocks on chunked data
AsyncHttpClient blocks while getting a chunked stream. Some sources need to process incoming data "on the fly" as the stream is received, e.g. unpack a sequence of JSON objects:
 
```nim
import asyncdispatch
import httpclient
var client = newAsyncHttpClient()

proc main() {.async.} =
  let r = await client.get "http://stream.meetup.com/2/rsvps"
```

### PR DESCRIPTION
**Pull request by ehmry**
Store the body completion future at the client and wait for it to
complete before issuing additional requests. This allows the body
FutureStream reader to drain the stream and read buffers to be freed
asynchronously.

Fix #8109

### CODE REVIEW

**Comment by ehmry** [comment]
My problem here was that I would run out of memory when downloading large files, the entire file would buffer in the response `FutureStream` before writing to file. I would also like to use the httpclient module for audio streams, but the same issue, the body stream is not dequed until completion.

See #8109

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -1159,6 +1164,12 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url: string,
   # Helper that actually makes the request. Does not handle redirects.
   let requestUrl = parseUri(url)
 
+  when client is AsyncHttpClient:
+    if (not client.pending.isNil) and (not client.pending.finished):
```
Comment (dom96) [inline lib/pure/httpclient.nim]: ``and (not client.pending.finished)`` isn't necessary. ``await`` will be a no-op if `pending` is finished.

File: lib/pure/httpclient.nim [hunk 2]
```diff
@@ -807,6 +807,7 @@ type
     lastProgressReport: float
     when SocketType is AsyncSocket:
       bodyStream: FutureStream[string]
+      pending: Future[void]
```
Comment (dom96) [inline lib/pure/httpclient.nim]: Nitpick but I would prefer if this was called ``parseBodyFut``

**Comment by ehmry** [comment]
@dom96, thats reasonable to me, updated.

**Comment by dom96** [comment]
Thanks!

**Pull request merged**